src/animation/mod.rs

```diff
@@ -259,6 +262,15 @@ impl AnimationPlayer2D {
         self.animation.elapsed
     }
 
+    /// Duration of the playing animation
+    ///
+    /// Note: This is independent of speed.
+    /// Note: This may be `0.0` right after setting the animation, make sure to handle that properly
+    /// e.g. when performing a division.
```

Comment:
Not really a fan of that.

Reply:
Yeah I also don't like it. I wrapped the return in an Option, I kinda not hate this, the user now has to explicitly handle the case when no duration is set. What do you think?

Reply:
Options sounds good